### PR TITLE
added sql db data maksing enabled

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -989,7 +989,8 @@ module.exports = {
         'eventHubMinimumTLSversion'     : require(__dirname + '/plugins/azure/eventhub/eventHubMinimumTLSversion.js'),
 
         'accessLogsEnabled'             : require(__dirname + '/plugins/azure/frontdoor/accessLogsEnabled.js'),
-        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
+        'frontDoorMinimumTlsVersion': require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js'),
+        'dbDataMaskingEnabled': require(__dirname + '/plugins/azure/sqldatabases/dbDataMaskingEnabled.js'),
```
Comment (mehakseedat63) [inline exports.js:993]: add it under other sqldatabases plugins

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Set Dynamic Data Masking for SQL Databases',
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:5]:
```suggestion
    title: 'Data Masking Enabled for SQL Databases',
```
change it to something like this or 'SQL Databases Data Masking Enabled'

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    category: 'SQL Databases',
+    domain: 'Databases',
+    description: 'Set up dynamic data masking to protect sensitive data exposure in SQL databases.',
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:8]:
```suggestion
    description: 'Ensures dynamic data masking is enabeld for all SQL databases.',
```

description should explain what the plugin does and not what should be done. 
'Set up dynamic data masking to protect sensitive data exposure in SQL databases.' should be in recommended action

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No SQL servers found', location);
+                return rcb();
+            }
+
+            // Loop through servers and check databases
+            servers.data.forEach(function(server) {
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:35]:
```suggestion
            servers.data.forEach(server => {
```

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+                if (!databases || databases.err || !databases.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for SQL server databases: ' + helpers.addError(databases), location, server.id);
+                } else {
+                    if (!databases.data.length) {
+                        helpers.addResult(results, 0,
+                            'No databases found for SQL server', location, server.id);
+                    } else {
+                        databases.data.forEach(function(database) {
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:47]:
```suggestion
                        databases.data.forEach(database =>  {
```

use arrow syntax

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                            
+                            var dataMaskingPolicies = helpers.addSource(cache, source, ['dataMaskingPolicies', 'get', location, database.id]);
+                            if (!dataMaskingPolicies || dataMaskingPolicies.err || !dataMaskingPolicies.data) {
+                                helpers.addResult(results, 3, 'Unable to query dynamic data masking: ' + helpers.addError(dataMaskingPolicies), location, database.id);
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:51]:
```suggestion
                                helpers.addResult(results, 3, 'Unable to query dynamic data masking policies: ' + helpers.addError(dataMaskingPolicies), location, database.id);
```

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                            } else {
+                                if (dataMaskingPolicies.data.dataMaskingState.toLowerCase()=='enabled') {
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:53]:
```suggestion
                                if (dataMaskingPolicies.data.dataMaskingState.toLowerCase()=='enabled') {
```
dataMaskingPolicies.data is not going to be an array? 
also check if dataMaskingState exists before converting to lowercase

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                                    helpers.addResult(results, 0, 'Dynamic data masking is enabled for the database', location, database.id);
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:54]:
```suggestion
                                    helpers.addResult(results, 0, 'Dynamic data masking is enabled for SQL database', location, database.id);
```
make it consistent with other plugin results for same category

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+                                } else {
+                                    helpers.addResult(results, 2, 'Dynamic data masking is not enabled for the database', location, database.id);
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:56]:
```suggestion
                                    helpers.addResult(results, 2, 'Dynamic data masking is not enabled for SQL database', location, database.id);
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: exports.js [hunk 10]
```diff
@@ -989,7 +990,7 @@ module.exports = {
         'eventHubMinimumTLSversion'     : require(__dirname + '/plugins/azure/eventhub/eventHubMinimumTLSversion.js'),
 
         'accessLogsEnabled'             : require(__dirname + '/plugins/azure/frontdoor/accessLogsEnabled.js'),
-        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
+        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js'),
```
Comment (alphadev4) [inline exports.js:993]:
```suggestion
        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
```

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+                        'Unable to query for SQL server databases: ' + helpers.addError(databases), location, server.id);
+                } else {
+                    if (!databases.data.length) {
+                        helpers.addResult(results, 0,
+                            'No databases found for SQL server', location, server.id);
+                    } else {
+                        databases.data.forEach(database =>  {
+                            
+                            var dataMaskingPolicies = helpers.addSource(cache, source, ['dataMaskingPolicies', 'get', location, database.id]);
+                            if (!dataMaskingPolicies || dataMaskingPolicies.err || !dataMaskingPolicies.data || !dataMaskingPolicies.data.dataMaskingState) {
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:50]:
```suggestion
                            if (!dataMaskingPolicies || dataMaskingPolicies.err || !dataMaskingPolicies.data || !dataMaskingPolicies.data) {
```

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+                                helpers.addResult(results, 3, 'Unable to query dynamic data masking policies: ' + helpers.addError(dataMaskingPolicies), location, database.id);
+                            } else {
+                                if (dataMaskingPolicies.data.dataMaskingState.toLowerCase()=='enabled') {
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:53]:
```suggestion
                                if (dataMaskingPolicies.data.dataMaskingState && dataMaskingPolicies.data.dataMaskingState.toLowerCase()=='enabled') {
```

**Review by alphadev4 — APPROVED** [review]

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+                    helpers.addResult(results, 3,
+                        'Unable to query for SQL server databases: ' + helpers.addError(databases), location, server.id);
+                } else {
+                    if (!databases.data.length) {
+                        helpers.addResult(results, 0,
+                            'No databases found for SQL server', location, server.id);
+                    } else {
+                        databases.data.forEach(database =>  {
+                            
+                            var dataMaskingPolicies = helpers.addSource(cache, source, ['dataMaskingPolicies', 'get', location, database.id]);
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:49]:
```suggestion
                            var dataMaskingPolicies = helpers.addSource(cache, source, ['dataMaskingPolicies', 'get', location, database.id]);
                            
```

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js (hunk 2 above)
Comment (alphadev4) [reply plugins/azure/sqldatabases/dbDataMaskingEnabled.js]: or SQL Database Dynamic Data Maskin

File: plugins/azure/sqldatabases/dbDataMaskingEnabled.js [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+                            
+                            if (!dataMaskingPolicies || dataMaskingPolicies.err || !dataMaskingPolicies.data || !dataMaskingPolicies.data) {
+                                helpers.addResult(results, 3, 'Unable to query dynamic data masking policies: ' + helpers.addError(dataMaskingPolicies), location, database.id);
+                            } else {
+                                if (dataMaskingPolicies.data.dataMaskingState && dataMaskingPolicies.data.dataMaskingState.toLowerCase()=='enabled') {
```
Comment (mehakseedat63) [inline plugins/azure/sqldatabases/dbDataMaskingEnabled.js:54]:
```suggestion
                                if (dataMaskingPolicies.data.dataMaskingState && dataMaskingPolicies.data.dataMaskingState.toLowerCase() == 'enabled') {
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**